frontend/src/metabase/dashboard/hooks/use-dashboard-fullscreen.ts

```diff
@@ -3,8 +3,10 @@ import { useCallback, useEffect, useState } from "react";
 
 import type { DashboardFullscreenControls } from "../types";
 
-export const useDashboardFullscreen = (): DashboardFullscreenControls => {
-  const { toggle, fullscreen } = useFullscreen();
+export const useDashboardFullscreen = (): DashboardFullscreenControls & {
+  ref: (element: HTMLElement | null) => void;
+} => {
+  const { ref, toggle, fullscreen } = useFullscreen();
```

Comment:
we can use Mantine's built in `ref` to have a decent fullscreen experience for dashboards